src/box/lua/config/source/file.lua

```diff
@@ -26,6 +26,10 @@ function methods.sync(self, config_module, _iconfig)
             config_module._config_file, res))
     end
 
+    -- YAML returns `nil` on empty file, while config sources
+    -- should be {} if empty.
+    res = res or {}
```

Comment:
Let's consider the following possible situations.

1. No config file.
2. An empty config file.
3. A config file with an empty YAML document (`---\n...\n`).
4. A config file with YAML's representation of nil/null value (`null` or `--- null\n...\n`).

Before this patch we handle 1) correctly, but other leads to a non-descriptive error. Now, 2) and 3) are handled correctly too.

However, 4) gives `box.NULL` from `yaml.decode()`, so `or {}` doesn't work.

```lua
tarantool> yaml.decode(yaml.encode(nil)) or {}
---
- null
...
```

I propose to handle `box.NULL` here in the same way as `nil`:

```lua
if res == nil then
    res = {}
end
```

This way we cover 4) in the same way as 1), 2) and 3).

Reply:
Actually, it seems case 3 isn't working properly as well, YAML treats it as an empty string and returns such, so the check should be:
```lua
if res == nil or res == box.NULL or res == ""
    res = {}
end
```

Reply:
Wow, I forgot about this. Found my own commit 4095e3050a82223b138a71f1723ffb320f6ec142, where I've commented this behavior as tarantool specific one.

Personally I think that we shouldn't fix yaml decoding in configuration sources. But it would be good to interpret `nil` and `box.NULL` values from decoding in the same way.

BTW,

```lua
if res == nil or res == box.NULL then
    <...>
end

-- ..is the same as..

if res == nil then
    <...>
end
```

So, finally, I would leave it as follows.

```lua
if res == nil then
    res = {}
end
```

Reply:
Okay, fixed.